modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1166,29 +1304,38 @@ synchronized void stopTimers() {
          * Returns a set of nodes that should be added to zone's data nodes.
          *
          * @param scaleUpRevision Last revision of the scale up event.
-         * @param scaleDownRevision Last revision of the scale down event.
          * @param revision Revision of the event for which this data nodes is needed.
-         * @return Set of nodes that should be added to zone's data nodes.
+         * @return List of nodes that should be added to zone's data nodes.
          */
-        Set<String> nodesToAddToDataNodes(long scaleUpRevision, long scaleDownRevision, long revision) {
+        List<String> nodesToAddToDataNodes(long scaleUpRevision, long revision) {
             Long toKey = topologyAugmentationMap.floorKey(revision);
 
+            Long fromKey = topologyAugmentationMap.ceilingKey(scaleUpRevision);
```

Comment:
To be honest, I didn't get the idea. We have already agreed on 
```
acc([ceil(ms.scaleUp), floor(timer.scaleUp)])
accDown([ceil(ms.scaleDown), floor(timer.scaleDown)])
``` 
where we have square braces. 

I've checked `topologyAugmentationMap.subMap(scaleUpRevision, false, revision, true)` and there are a lot of failed tests.

Reply:
First of all 
`topologyAugmentationMap.subMap(scaleUpRevision, true, revision, true)` is the same as 
`topologyAugmentationMap.subMap(topologyAugmentationMap.ceilingKey(scaleUpRevision), true, topologyAugmentationMap.floorKey(revision), true)`
so `nodesToAddToDataNodes(long scaleUpRevision, long revision)` is the same as 
`accumulateNodes(scaleUpRevision, revision, true)`

Besides that
subMap(scaleUpRevision, **false** should work and it's actually proper one because you should exclude already accumulated values and left border already included in previous timer.